llvm/lib/Object/DXContainer.cpp

```diff
@@ -288,6 +289,50 @@ Error DirectX::RootSignature::parse(StringRef Data) {
                             llvm::Twine(FValue));
   Flags = FValue;
 
+  Current = Begin + RootParametersOffset;
+  for (uint32_t It = 0; It < NumParameters; It++) {
+    dxbc::RootParameter NewParam;
+
+    dxbc::RootParameterType PTValue =
+        support::endian::read<dxbc::RootParameterType,
+                              llvm::endianness::little>(Current);
+    if (!dxbc::RootSignatureValidations::isValidParameterType(PTValue))
+      return validationFailed("unsupported parameter type value read: " +
+                              llvm::Twine((uint32_t)PTValue));
+
+    NewParam.ParameterType =
+        support::endian::read<dxbc::RootParameterType,
+                              llvm::endianness::little>(Current);
```

Comment:
(I may not understand how `read` works, but couldn't this just be:

```suggestion
    NewParam.ParameterType = PTValue;
```

And if so, why do we need `PTValue` at all?  Could we just read directly into `NewParam.ParameterType` on line 296?

Reply:
Sorry, copy, paste and not update